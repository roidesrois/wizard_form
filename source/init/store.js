// Core
import { createStore } from 'redux';
import { connectRouter } from 'connected-react-router';

// Roots
import { rootReducer } from './rootReducer';
import { rootSaga } from './rootSaga';

// Middleware
import { enhancedStore, sagaMiddleware, history } from './middleware/core';

const store = createStore(connectRouter(history)(rootReducer), enhancedStore);

sagaMiddleware.run(rootSaga);

export { store, history };
