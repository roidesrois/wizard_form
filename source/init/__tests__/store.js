// Store
import { store } from '../store';

describe('store:', () => {
    test('should have a valid state shape', () => {
        expect(store.getState()).toMatchSnapshot();
    });
});
