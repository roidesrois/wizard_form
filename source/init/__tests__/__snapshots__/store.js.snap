// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store: should have a valid state shape 1`] = `
Object {
  "auth": Immutable.Map {
    "isInitialized": false,
    "isAuthenticated": false,
  },
  "forms": Object {
    "forms": Object {
      "$form": Object {
        "errors": Object {},
        "focus": false,
        "initialValue": Object {
          "user": Object {
            "profile": Object {
              "avatar": Array [],
              "firstName": "",
              "lastName": "",
            },
          },
        },
        "intents": Array [],
        "model": "forms",
        "pending": false,
        "pristine": true,
        "retouched": false,
        "submitFailed": false,
        "submitted": false,
        "touched": false,
        "valid": true,
        "validated": false,
        "validating": false,
        "validity": Object {},
        "value": Object {
          "user": Object {
            "profile": Object {
              "avatar": Array [],
              "firstName": "",
              "lastName": "",
            },
          },
        },
      },
      "user": Object {
        "$form": Object {
          "errors": Object {},
          "focus": false,
          "initialValue": Object {
            "profile": Object {
              "avatar": Array [],
              "firstName": "",
              "lastName": "",
            },
          },
          "intents": Array [],
          "model": "forms.user",
          "pending": false,
          "pristine": true,
          "retouched": false,
          "submitFailed": false,
          "submitted": false,
          "touched": false,
          "valid": true,
          "validated": false,
          "validating": false,
          "validity": Object {},
          "value": Object {
            "profile": Object {
              "avatar": Array [],
              "firstName": "",
              "lastName": "",
            },
          },
        },
        "profile": Object {
          "$form": Object {
            "errors": Object {},
            "focus": false,
            "initialValue": Object {
              "avatar": Array [],
              "firstName": "",
              "lastName": "",
            },
            "intents": Array [],
            "model": "forms.user.profile",
            "pending": false,
            "pristine": true,
            "retouched": false,
            "submitFailed": false,
            "submitted": false,
            "touched": false,
            "valid": true,
            "validated": false,
            "validating": false,
            "validity": Object {},
            "value": Object {
              "avatar": Array [],
              "firstName": "",
              "lastName": "",
            },
          },
          "avatar": Object {
            "$form": Object {
              "errors": Object {},
              "focus": false,
              "initialValue": Array [],
              "intents": Array [],
              "model": "forms.user.profile.avatar",
              "pending": false,
              "pristine": true,
              "retouched": false,
              "submitFailed": false,
              "submitted": false,
              "touched": false,
              "valid": true,
              "validated": false,
              "validating": false,
              "validity": Object {},
              "value": Array [],
            },
          },
          "firstName": Object {
            "errors": Object {},
            "focus": false,
            "initialValue": "",
            "intents": Array [],
            "model": "forms.user.profile.firstName",
            "pending": false,
            "pristine": true,
            "retouched": false,
            "submitFailed": false,
            "submitted": false,
            "touched": false,
            "valid": true,
            "validated": false,
            "validating": false,
            "validity": Object {},
            "value": "",
          },
          "lastName": Object {
            "errors": Object {},
            "focus": false,
            "initialValue": "",
            "intents": Array [],
            "model": "forms.user.profile.lastName",
            "pending": false,
            "pristine": true,
            "retouched": false,
            "submitFailed": false,
            "submitted": false,
            "touched": false,
            "valid": true,
            "validated": false,
            "validating": false,
            "validity": Object {},
            "value": "",
          },
        },
      },
    },
    "user": Object {
      "profile": Object {
        "avatar": Array [],
        "firstName": "",
        "lastName": "",
      },
    },
  },
  "posts": Immutable.List [],
  "profile": Immutable.Map {
    "id": "",
    "avatar": "",
    "firstName": "",
    "lastName": "",
    "token": "",
  },
  "router": Object {
    "action": "POP",
    "location": Object {
      "hash": "",
      "pathname": "/",
      "search": "",
      "state": undefined,
    },
  },
  "ui": Immutable.Map {
    "notification": Immutable.Map {},
    "isOnline": false,
    "isFetching": false,
  },
  "users": Immutable.List [],
}
`;
